libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -33,13 +35,15 @@ public class CosmosDbStorage : IStorage
         private readonly RequestOptions _databaseCreationRequestOptions = null;
         private readonly IDocumentClient _client;
         private string _collectionLink = null;
+        private readonly AzureServiceTokenProvider _tokenProvider;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CosmosDbStorage"/> class.
         /// using the provided CosmosDB credentials, database ID, and collection ID.
         /// </summary>
         /// <param name="cosmosDbStorageOptions">Cosmos DB storage configuration options.</param>
-        public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions)
+        /// <param name="tokenProvider">optional token provider to get token to connect to Cosmos DB</param>
+        public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions, AzureServiceTokenProvider tokenProvider)
```

Comment:
>AzureServiceTokenProvider tokenProvider [](start = 78, length = 39)

This is a breaking change, you can't change the signature like this.
Can you refactor to have ctor overload or make tokenProvider = null as default?

Reply:
actually, I can revert this change, and use the injectable IDocumentClient instead